core/src/main/scala/org/http4s/ContextRequest.scala

```diff
@@ -0,0 +1,20 @@
+package org.http4s
+
+import cats.syntax.functor._
+import cats.{Functor, ~>}
+import cats.data.Kleisli
+
+final case class ContextRequest[F[_], A](context: A, req: Request[F]) {
+  def mapK[G[_]](fk: F ~> G): ContextRequest[G, A] =
+    ContextRequest(context, req.mapK(fk))
+
+  @deprecated("Use context instead", "0.21.0")
+  def authInfo: A = context
```

Comment:
Seeing how many places this changed just in this repo, it would be a great candidate for a scalafix, but that's not essential to this PR.